### PR TITLE
Add a new rule for sending email

### PR DESCRIPTION
**Pull request by seamys**
@SebastianStehle 

Add a new rule for sending email.

### CODE REVIEW

File: extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+namespace Squidex.Extensions.Actions.Email
+{
+    public sealed class EmailActionHandler : RuleActionHandler<EmailAction, EmailJob>
+    {
+        private const string Description = "Send an Email";
+
+        private readonly EmailOptions emailOptions;
+
+        public EmailActionHandler(RuleEventFormatter formatter, IOptions<EmailOptions> emailOptions)
```
Comment (SebastianStehle) [inline extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs]: I would add the email options to the action, because it cannot be used with the cloud otherwise.

File: extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        {
+            Guard.NotNull(emailOptions, nameof(emailOptions));
+            this.emailOptions = emailOptions.Value;
+        }
+
+        protected override (string Description, EmailJob Data) CreateJob(EnrichedEvent @event, EmailAction action)
+        {
+            var ruleJob = new EmailJob
+            {
+                From = action.From,
```
Comment (SebastianStehle) [inline extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs]: I think from and to should be also formattable, e.g. you could create the email to the user who created a content.

File: extensions/Squidex.Extensions/Actions/Email/EmailAction.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+using Squidex.Domain.Apps.Core.Rules;
+
+namespace Squidex.Extensions.Actions.Email
+{
+    [RuleActionHandler(typeof(EmailActionHandler))]
+    [RuleAction(
+    IconImage = "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32'><path d='M28 5h-24c-2.209 0-4 1.792-4 4v13c0 2.209 1.791 4 4 4h24c2.209 0 4-1.791 4-4v-13c0-2.208-1.791-4-4-4zM2 10.25l6.999 5.25-6.999 5.25v-10.5zM30 22c0 1.104-0.898 2-2 2h-24c-1.103 0-2-0.896-2-2l7.832-5.875 4.368 3.277c0.533 0.398 1.166 0.6 1.8 0.6 0.633 0 1.266-0.201 1.799-0.6l4.369-3.277 7.832 5.875zM30 20.75l-7-5.25 7-5.25v10.5zM17.199 18.602c-0.349 0.262-0.763 0.4-1.199 0.4s-0.851-0.139-1.2-0.4l-12.8-9.602c0-1.103 0.897-2 2-2h24c1.102 0 2 0.897 2 2l-12.801 9.602z'/></svg>",
+    IconColor = "#333300",
+    Display = "Send an email",
+    Description = "Send an email, when some contents has changed",
```
Comment (SebastianStehle) [inline extensions/Squidex.Extensions/Actions/Email/EmailAction.cs]: An action should not require a specific trigger, e.g. you could also send a content when an asset has changed.

File: src/Squidex/app/features/rules/pages/rules/actions/email-action.component.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+@import '_vars';
+@import '_mixins';
+textarea {
```
Comment (SebastianStehle) [inline src/Squidex/app/features/rules/pages/rules/actions/email-action.component.scss]: Please add an empty line here.

File: extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                To = action.To,
+                Subject = Format(action.Subject, @event),
+                Body = Format(action.Body, @event)
+            };
+            return (Description, ruleJob);
+        }
+
+        protected override async Task<(string Dump, Exception Exception)> ExecuteJobAsync(EmailJob job)
+        {
+            using (SmtpClient client = new SmtpClient(emailOptions.Host, emailOptions.Port))
```
Comment (SebastianStehle) [inline extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs]: Please use var whenever possible.

File: extensions/Squidex.Extensions/Actions/Email/EmailAction.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+// ==========================================================================
+
+using System.ComponentModel.DataAnnotations;
+using Squidex.Domain.Apps.Core.Rules;
+
+namespace Squidex.Extensions.Actions.Email
+{
+    [RuleActionHandler(typeof(EmailActionHandler))]
+    [RuleAction(
+    IconImage = "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32'><path d='M28 5h-24c-2.209 0-4 1.792-4 4v13c0 2.209 1.791 4 4 4h24c2.209 0 4-1.791 4-4v-13c0-2.208-1.791-4-4-4zM2 10.25l6.999 5.25-6.999 5.25v-10.5zM30 22c0 1.104-0.898 2-2 2h-24c-1.103 0-2-0.896-2-2l7.832-5.875 4.368 3.277c0.533 0.398 1.166 0.6 1.8 0.6 0.633 0 1.266-0.201 1.799-0.6l4.369-3.277 7.832 5.875zM30 20.75l-7-5.25 7-5.25v10.5zM17.199 18.602c-0.349 0.262-0.763 0.4-1.199 0.4s-0.851-0.139-1.2-0.4l-12.8-9.602c0-1.103 0.897-2 2-2h24c1.102 0 2 0.897 2 2l-12.801 9.602z'/></svg>",
```
Comment (SebastianStehle) [inline extensions/Squidex.Extensions/Actions/Email/EmailAction.cs]: Please have a look to the formatting of other actions.

File: extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        private const string Description = "Send an Email";
+
+        private readonly EmailOptions emailOptions;
+
+        public EmailActionHandler(RuleEventFormatter formatter, IOptions<EmailOptions> emailOptions)
+            : base(formatter)
+        {
+            Guard.NotNull(emailOptions, nameof(emailOptions));
+            this.emailOptions = emailOptions.Value;
```
Comment (SebastianStehle) [inline extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs]: Please add an empty line here.

File: extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs (hunk 1 above)
Comment (seamys) [reply extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs]: Yes, good idea. i will move those options to action, thanks

File: extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                }
+            }
+
+            return ("Completed", null);
+        }
+    }
+
+    public class EmailJob
+    {
+        public string Host { get; set; }
```
Comment (SebastianStehle) [inline extensions/Squidex.Extensions/Actions/Email/EmailActionHandler.cs]: It is a personal preference, but if I have so much properties I tend to group them with a prefix, e.g. ServerHost, ServerUseSsl, ServerUsername, MessageFrom ...

File: src/Squidex/app/features/rules/pages/rules/actions/email-action.component.html [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+    <div class="form-group row">
+        <label class="col-3 col-form-label" for="from">From</label>
+
+        <div class="col-9">
+            <sqx-control-errors for="from" [submitted]="actionFormSubmitted"></sqx-control-errors>
+
+            <input type="text" class="form-control" id="from" formControlName="from" />
+
+            <small class="form-text text-muted">
+                The email is sent from?
```
Comment (SebastianStehle) [inline src/Squidex/app/features/rules/pages/rules/actions/email-action.component.html]: Please do not use questions for descriptions. In general it is fine for me, but it is not used anywhere else.

File: src/Squidex/app/features/rules/pages/rules/actions/email-action.component.html [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+    <div class="form-group row">
+        <label class="col-3 col-form-label" for="body">Body</label>
+
+        <div class="col-9">
+            <sqx-control-errors for="body" [submitted]="actionFormSubmitted"></sqx-control-errors>
+
+            <textarea class="form-control" id="body" formControlName="body"></textarea>
+
+            <small class="form-text text-muted">
+                The message body
```
Comment (SebastianStehle) [inline src/Squidex/app/features/rules/pages/rules/actions/email-action.component.html]: If you have a look to the other hints you will that I have added a link to the docs for formatting action inputs. Can you add it too?

File: src/Squidex/appsettings.json [hunk 11]
```diff
@@ -68,8 +68,8 @@
       /*
```
Comment (SebastianStehle) [inline src/Squidex/appsettings.json]: Can you revert the changes in this file?
